### PR TITLE
RFC: providers/dns: add Service Account authorization option to gcloud

### PR DESCRIPTION
**Pull request by danrl**
Using a Service Account file allows us to operate easier in an automated environment. Service Account files can be obtained from GCP IAM. If the GCE_SERVICE_ACCOUNT_FILE env var is set, it is used for authentication.

We may discuss a fallback to NewDNSProviderCredentials if NewDNSProviderServiceAccount fails.

### CODE REVIEW

**Comment by danrl** [comment]
Updated to branch to reflect the latest master.

Any comments on the PR?

File: providers/dns/googlecloud/googlecloud.go [hunk 1]
```diff
@@ -22,9 +24,15 @@ type DNSProvider struct {
...
 
 // NewDNSProvider returns a DNSProvider instance configured for Google Cloud
-// DNS. Credentials must be passed in the environment variable: GCE_PROJECT.
+// DNS. Project name must be passed in the environment variable: GCE_PROJECT.
+// A Service Account file can be passed in the environment variable:
+// GCE_SERVICE_ACCOUNT_FILE
 func NewDNSProvider() (*DNSProvider, error) {
 	project := os.Getenv("GCE_PROJECT")
+	saFile := os.Getenv("GCE_SERVICE_ACCOUNT_FILE")
+	if len(saFile) > 0 {
```
Comment (xenolf) [inline providers/dns/googlecloud/googlecloud.go]: I would replace this whole check with a call to `os.LookupEnv` as the value is only used within the if block.

**Comment by xenolf** [comment]
Thank you for your time working on this PR!

> We may discuss a fallback to NewDNSProviderCredentials if NewDNSProviderServiceAccount fails.

Lego defaults to failing with a helpful error message if a configuration parameter was faulty, I think we should keep it like that.

File: providers/dns/googlecloud/googlecloud.go (hunk 1 above)
Comment (danrl) [inline providers/dns/googlecloud/googlecloud.go]: Thanks for the review! I changed it as requested.

Comment (xenolf) [inline providers/dns/googlecloud/googlecloud.go]: Thanks for the quick change! One more nitpick 😅 
`if saFile, ok := os.LookupEnv("GCE_SERVICE_ACCOUNT_FILE"); ok {
}`

Comment (danrl) [inline providers/dns/googlecloud/googlecloud.go]: Done via fixup and rebase :)

**Comment by danrl** [comment]
Thx! Awesome!

**Pull request merged**